parlai/tasks/anli/agents.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import copy
+import json
+
+from parlai.core.teachers import DialogTeacher, MultiTaskTeacher
+from .build import build
+
+from .anli_constants import ANLI, ANLI_VERSION, ANLI_ANSWER_KEY, ANLI_LABEL_DICT, ANLI_HYPO_KEY, \
+    ANLI_HYPO_PREFIX, ANLI_LABELS, ANLI_PREFIX, ANLI_PREMISE_KEY, ANLI_PREMISE_PREFIX, ANLI_ROUNDS
+
+
+def _path(opt):
+    build(opt)
+
+    dt = opt['datatype'].split(':')[0]
+
+    if dt == 'train':
+        suffix = 'train'
+    elif dt == 'valid':
+        suffix = 'dev'
+    elif dt == 'test':
+        suffix = 'test'
+    else:
+        raise RuntimeError('Not valid datatype.')
+
+    data_path = os.path.join(
+        opt['datapath'],
+        ANLI,
+        ANLI_PREFIX + ANLI_VERSION,
+        opt['round'],
+        suffix + '.jsonl',
+    )
+
+    return data_path
+
+
+def setup_data(path):
```

Comment:
go ahead and move this into the teacher, it doesn't need to be a separate method.